app/scripts/services/dimBungieService.factory.js

```diff
@@ -68,6 +71,27 @@
                  response.config.url.indexOf('/Account/') >= 0 &&
                  response.config.url.indexOf('/Character/') < 0) {
         return $q.reject(new Error('No Destiny account was found for this platform.'));
+      } else if (errorCode === 2107 || errorCode === 2101 || errorCode === 2102) {
+        if (dimSettingsService.apiKey) {
```

Comment:
thinking about it now, it should just use native `localStorage` instead of bothering with `dimSettingsService`